source/getting-started/java.md

```diff
@@ -68,312 +66,160 @@ See also: [app authorization in the API reference](/reference/#authorizing-your-
-    	// do something            
-	}
-
-	@Override
-	public void onCacheError(String errorMessage) {
-		// do something
-	}
-
-	@Override
-	public void apiCallback(List<Event> events, Map<String, Double> eventDeletions, Double serverTime) {
-		// do something
-	}
-
-	@Override
-	public void onApiError(String errorMessage, Double serverTime) {
-		// do something
-	}
-});
+try {
+	Filter filter = new Filter().addStream('diary');
+	List<Event> retrievedEvents = connection.events.get(filter);
+	// Do something with the retrieved Events
+} catch (IOException e) {
+	// Handle the error
+}
 ```
 
 #### Create
 
 ```java
-Event newEvent = new Event()
-newEvent.setStreamId("diary");
-newEvent.setType("note/txt");
-newEvent.setContent("I track, therefore I am.");
-connection.events.create(newEvent, new EventsCallback() {
-	@Override
-	public void onApiSuccess(String successMessage, Event event, String stoppedId, Double serverTime) {
-    	// do something            
-	}
-
-	@Override
-	public void onApiError(String errorMessage, Double serverTime) {
-		// do something
-	}
-
-	@Override
-	public void onCacheSuccess(String successMessage, Event event) {
-		// do something
-	}
-
-	@Override
-	public void onCacheError(String errorMessage) {
-		// do something
-	}
-});
+try {
+	Event newEvent = new Event()
+		.setStreamId("diary")
+		.setType("note/txt")
+		.setContent("I track, therefore I am.");
+	newEvent = connection.events.create(newEvent);
+	// Do something with the created Event
+} catch (IOException e) {
+	// Handle the error
+}
 ```
 
 #### Update
 
 ```java
-event.setContent = "Updated content.";
-connection.events.update(event, new EventsCallback() {
-	@Override
-	public void onApiSuccess(String successMessage, Event event, String stoppedId, Double serverTime) {
-    	// do something            
-	}
-
-	@Override
-	public void onApiError(String errorMessage, Double serverTime) {
-		// do something
-	}
-
-	@Override
-	public void onCacheSuccess(String successMessage, Event event) {
-		// do something
-	}
-
-	@Override
-	public void onCacheError(String errorMessage) {
-		// do something
-	}
-});
+try {
+	newEvent.setContent("updated content");
+	Event updatedEvent = connection.events.update(newEvent);
+	// Do something with the updated Event
+} catch (IOException e) {
+	// Handle the error
+}
 ```
 
 #### Delete
 
 ```java
-connection.events.delete(event, new EventsCallback() {
-	@Override
-	public void onApiSuccess(String successMessage, Event event, String stoppedId, Double serverTime) {
-    	// do something            
-	}
-
-	@Override
-	public void onApiError(String errorMessage, Double serverTime) {
-		// do something
-	}
-
-	@Override
-	public void onCacheSuccess(String successMessage, Event event) {
-		// do something
-	}
-
-	@Override
-	public void onCacheError(String errorMessage) {
-		// do something
-	}
-});
+try {
+	String eventDeletionId = connection.events.delete(newEvent.getId());
```

Comment:
You still return an ID in case of trash. Not cool

Reply:
This will be part of 2.1.0 (alongside with the specific Exceptions)